tools/doc/json.js

```diff
@@ -272,21 +272,30 @@ function parseSignature(text, sig) {
   var params = text.match(paramExpr);
   if (!params) return;
   params = params[1];
-  // the [ is irrelevant. ] indicates optionalness.
-  params = params.replace(/\[/g, '');
   params = params.split(/,/);
+  var optionalLevel = 0;
+  var optionalCharDict = {'[': 1, ' ': 0, ']': -1};
   params.forEach(function(p, i, _) {
     p = p.trim();
     if (!p) return;
     var param = sig.params[i];
     var optional = false;
     var def;
-    // [foo] -> optional
-    if (p.charAt(p.length - 1) === ']') {
-      optional = true;
-      p = p.replace(/\]/g, '');
-      p = p.trim();
+
+    // for grouped optional params such as someMethod(a[, b[, c]])
+    var pos;
+    for (pos = 0; pos < p.length; pos++) {
+      if ('[ ]'.indexOf(p[pos]) === -1) { break; }
```

Comment:
You can use `optionalCharDict` itself, to check if the current char is in it or not.